pkg/cdc/const.go

```diff
@@ -0,0 +1,3 @@
+package cdc
+
+var CDCAgentVer = "2.0.0"
```

Comment:
Lets not put versions to .go files, it requires the user to deploy a version of cass-operator whenever we need to change this version.

Reply:
This affects the path to the change agent .jar file so it is kind of required... What's your alternative? Would you change management API image builds to use the same path for CDC agent regardless of version?

I could be persuaded, but given that I think CDC agent is still a bit unstable, there are some benefits to tightly coupling the CDC agent version to cass-operator. My concern is in the following case:

* Change agent exposes an "API" via Java CLI arguments. 
* cass-operator embeds that API [here](https://github.com/Miles-Garnsey/cass-operator/blob/feature/cdc-integration/apis/cassandra/v1beta1/cdc_types.go#L5).
* If the change agent's API evolves, cass-operator will pass the wrong arguments via the reflection process [here](https://github.com/Miles-Garnsey/cass-operator/blob/feature/cdc-integration/pkg/cdc/generate_config.go#L57).

I think given where the maturity level of the technology is at, and given we don't understand the "API" evolution strategy for change agent (i.e. what arguments it will accept), I'd rather play safe and couple tightly.

Reply:
There's multiple issues in that approach though. Lets say there's a bug in 2.0.0. What will we ship in the management-api images? Are all those previous Cassandra versions now unable to update management-api because we can't anymore process them? Lets say 4.0.4 mgmt-api CDC has a bug. We can't update that 4.0.4 image, it has to ship with 2.0.0 forever - otherwise you will kill existing installations if they restart a pod. 

We can't ship 4.0.5 with 2.0.1 either, since now their cass-operator installation can no longer update their Cassandra to another minor version, so all of 4.0.* has to stay in 2.0.0 forever. And probably 4.1 also at this point. 

And if we update here to 2.0.1, that version of cass-operator can no longer create any new 4.0.x or 4.1.x installations.

And then it gets worse if we actually have to ship multiple Cassandra images. :4.0.4-2.0.1 tag and manually forcing the image for every CDC? User upgrades cass-operator, but that cass-operator might require an update to Kubernetes also. Thus, their 2.0.0 -> 2.0.1 might require both Kubernetes as well as Cassandra major version changes.

Reply:
> This affects the path to the change agent .jar file so it is kind of required... What's your alternative? Would you change management API image builds to use the same path for CDC agent regardless of version?

@Miles-Garnsey We may want to alter the Management API image so that it removes the version number from the CDC agent jarfile so that the version can change and cass-operator doesn't have to know or care about specific versions. If you decide you want to do this, we can cut a new Management API release pretty quickly. Talking about [this line](https://github.com/k8ssandra/management-api-for-apache-cassandra/blob/master/Dockerfile-4_0#L40) in the Dockerfile.

Reply:
I can see your point here. It does seem that tightly coupling in the way I'm proposing will force a (possibly unnecessary) operator version upgrade whenever we bump a CDC agent version.

The flip side is that, if we make this change, the API we are using may change which could cause hard-to-detect failures.

My proposal would still lead to a failure in management API (at Cassandra start) which still isn't where this error is actually coming from (it comes from cass-operator starting Cassandra with the wrong Java parms).

On balance, I'm actually going to support the proposal that we DO change management API's image builds to always place CDC agent in the same location regardless of version. The implications are that management API and cass-operator become agnostic as to the CDC agent version in use.

If we subsequently find that the CDC agent API does change, we can build additional validation to catch and communicate this to the user in a clearer way.